### PR TITLE
MOD-5965: Return error upon timeout

### PR DESCRIPTION
**Pull request by raz-mon**
Currently, in case we experience a timeout throughout pipeline execution, we may reply to the client with a simple string depicting the error message, instead of replying with an error (with that same error message).

This PR fixes this, so that we always return an error.

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/aggregate/aggregate_exec.c [hunk 1]
```diff
@@ -345,7 +345,7 @@ void sendChunk(AREQ *req, RedisModule_Reply *reply, size_t limit) {
 
     RedisModule_ReplyKV_Array(reply, "error"); // >errors
       if (rc == RS_RESULT_TIMEDOUT) {
-        RedisModule_Reply_SimpleString(reply, "Timeout limit was reached");
+        RedisModule_Reply_Error(reply, QueryError_Strerror(QUERY_TIMEDOUT));
```
Comment (nafraf) [inline src/aggregate/aggregate_exec.c:348]: Is this line covered this line by the tests?
I see the `test_resp3:test_search_timeout` but codecov reported this line as uncovered.

Comment (raz-mon) [reply src/aggregate/aggregate_exec.c:348]: The test in `test_resp3:test_search_timeout` is failing before, in `prepareExecutionPlan` (timed out there).
I'll add a test failing here.

**Review by nafraf — APPROVED** [review]
LGTM

File: tests/pytests/test_resp3.py [hunk 2]
```diff
@@ -162,10 +162,22 @@ def test_search_timeout():
...
     env.expect('ft.search', 'myIdx', 'aa*|aa*|aa*|aa* aa*', 'limit', '0', '0'). \
-      contains('Timeout limit was reached')
+      error().contains('Timeout limit was reached')
     env.expect('ft.search', 'myIdx', 'aa*|aa*|aa*|aa* aa*', 'timeout', 1).\
       error().contains('Timeout limit was reached')
 
+    # (coverage) Later failure than the above tests - in pipeline execution
+    # phase. For this, we need more documents in the index, such that we will
+    # fail for sure
+    num_range_2 = 30000
```
Comment (oshadmi) [inline tests/pytests/test_resp3.py:172]: Can increase to be less flaky and add pipeline+execute

**Review by nafraf — DISMISSED** [review]

**Review by oshadmi — APPROVED** [review]
👍🏼

**Review by oshadmi — APPROVED** [review]

**Pull request merged**